posthog-android/src/main/java/com/posthog/android/internal/PostHogAndroidUtils.kt

```diff
@@ -144,6 +150,38 @@ internal fun Activity.activityLabelOrName(config: PostHogAndroidConfig): String?
     }
 }
 
+@Suppress("DEPRECATION")
+internal fun Activity.getReferrerInfo(config: PostHogAndroidConfig): Map<String, String>? {
+    return try {
+        val intent = this.intent
+        val referrerInfoMap = mutableMapOf<String, String>()
+        val referrer: Uri?
+        val referrerName: String?
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
+            referrer = intent.getParcelableExtra(Intent.EXTRA_REFERRER)
+            referrerName = intent.getStringExtra(Intent.EXTRA_REFERRER_NAME)
+        } else {
+            referrer = intent.getParcelableExtra("android.intent.extra.REFERRER")
+            referrerName = intent.getStringExtra("android.intent.extra.REFERRER_NAME")
+        }
+        if (referrer == null && referrerName == null) {
+            return null
+        } else if (referrer == null) {
+            Uri.parse(referrerName)?.let { uri ->
```

Comment:
does Uti.parse throw if its an invalid uri or just returns null?

Reply:
Oh, I made a mistake, Uri.parse does throw NullPointerException

Reply:
in this case its better to make an extension to Uri.parse that swallow the exception if its invalid/null instead of dropping all the data that could be captured (eg there's referrer but not referrerName and vice-versa)